File: repeated_string/rs.js
const repeatedString = (s, n) => {
	if(s === 'a'){
		return n
	}else{
		let re = /a/g;
		let repeat_a = s.match(re).length;
		let repeatStr = ~~(n / s.length);
		let totalA = (repeat_a * repeatStr) + s.slice(0, n%s.length).match(re).length;
		return totalA;
	}
}

console.log(repeatedString('aba', 10));
console.log(repeatedString('a', 1000000000000));

File: repeated_string/rs.py
def repeated_string(s:str, n:int)->int:
    if s == 'a':
        return n
    else:
        count_a = s.count('a')
        repeat_str = n//len(s)
        repeat_a = repeat_str * count_a + s[:(n%len(s))].count('a')
        return repeat_a

if __name__ == '__main__':
    print(repeated_string('aba', 10))
    print(repeated_string('a', 1000000000000))
